opensoundscape/localization.py

```diff
@@ -475,46 +538,54 @@ def localize_detections(
             detections,
             min_n_receivers,
             max_receiver_dist,
+            bandpass_ranges=bandpass_ranges,
+            cc_filter=cc_filter,
         )
 
-        # attempt to localize each event
-        for event in candidate_events:
-            # choose bandpass range based on this event's detected class
-            if bandpass_ranges is not None:
-                bandpass_range = bandpass_ranges[event.class_name]
-            else:
-                bandpass_range = None
-
-            # perform gcc to estimate relative time of arrival at each receiver
-            # relative to the first in the list (reference receiver)
-            event.estimate_delays(
-                bandpass_range=bandpass_range,
+        # perform gcc to estimate relative time of arrival at each receiver
+        # estimate locations of sound event using time delays and receiver locations
+        # this calls estimate_delays under the hood
+        # unparallelized below
+        # events = [
+        #     e._estimate_location_return_self(
+        #         algorithm=localization_algorithm,
+        #         cc_threshold=cc_threshold,
+        #         cc_filter=cc_filter,
+        #         min_n_receivers=min_n_receivers,
+        #         speed_of_sound=SPEED_OF_SOUND,
+        #         bandpass_range=bandpass_ranges[e.class_name],
+        #     )
+        #     for e in candidate_events
+        # ]
+
+        # # paralelize the above using joblib
+        from joblib import Parallel, delayed
+
+        events = Parallel(n_jobs=n_workers)(
```

Comment:
I have to admit I'm nervous about internally implementing parallelization. For instance, the .ml module uses PyTorch's parallelization but doesn't implement any, and the .ribbit module does not parallelize, the user must explicitly parallelize in a script. 

At the minimum, passing n_workers=0 or 1 should probably avoid using Parallel all together, so that the user can avoid Parallel if it causes any issues

Reply:
Also, could you look at this documentation and confirm that the default (multi-processing rather than multi-threading) is the right choice here? I think it probably is, but have a look
https://joblib.readthedocs.io/en/latest/parallel.html#thread-based-parallelism-vs-process-based-parallelism

Reply:
I am also nervous about parallelizing, but the cross-correlation process is very slow without it and we largely hide the internals from the user, so it would be hard for them to parallellize it themselves. I think the default of multiprocessing makes sense, because the slow part is the computation. It is not i/o that is limiting.